src/test/java/com/gui/HomePageTest.java

```diff
@@ -4,40 +4,52 @@
 import com.gui.pages.desktop.HomePage;
 import com.gui.pages.desktop.ProductsListPage;
 import org.testng.Assert;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 
 public class HomePageTest extends AbstractTest {
 
-    @Test
-    public void test(){
-        HomePage homePage =new HomePage(getDriver());
-        ProductsListPage productsListPage = homePage.searchKeyWord("shirt");
-        productsListPage.printAllItems();
-    }
 
-    @Test(groups = "group1")
+    @Test(groups = "group1", dataProvider = "currencyData")
     public void testCurrencyChange() {
-        HomePage homePage =new HomePage(getDriver());
+        HomePage homePage = new HomePage(getDriver());
         homePage.changeCurrency("£");
-        Assert.assertTrue(homePage.isCurrencyInCartCorrect("£"), "Currency is incorrect");
+        Assert.assertTrue(homePage.isCurrencyInCartCorrect("£"), "Currency in cart doesn't match the");
     }
 
-    @Test(groups = "group1")
-    public void testCartItemCount(){
-        HomePage homePage =new HomePage(getDriver());
+    @Test(groups = "group1", dataProvider = "itemCountData")
+    public void testCartItemCount() {
```

Comment:
you must use the parameters in the test method signature and inside the test logic.
@Test(groups = "group1", dataProvider = "currencyData")
public void testCurrencyChange(String currencyInput, String expectedCurrency) {
    HomePage homePage = new HomePage(getDriver());
    homePage.changeCurrency(currencyInput);
    Assert.assertTrue(homePage.isCurrencyInCartCorrect(expectedCurrency), "Currency is incorrect");
}